src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -606,6 +618,126 @@ private StorageIterator GetBlockedAccounts(DataCache snapshot)
+
+            var validatedTokens = new HashSet<UInt160>
+            {
+                NEO.Hash,
+                GAS.Hash
+            };
+
+            foreach (var tokenItem in extraTokens)
+            {
+                var span = tokenItem.GetSpan();
+                if (span.Length != UInt160.Length)
+                    throw new ArgumentException($"Invalid token hash length: expected {UInt160.Length} bytes, got {span.Length} bytes.");
+
+                var contractHash = new UInt160(span);
+
+                // Validate contract exists
+                var contract = ContractManagement.GetContract(engine.SnapshotCache, contractHash);
+                if (contract == null)
+                    throw new InvalidOperationException($"Contract {contractHash} does not exist.");
+
+                // Validate contract implements NEP-17 standard
+                if (!contract.Manifest.SupportedStandards.Contains("NEP-17"))
+                    throw new InvalidOperationException($"Contract {contractHash} does not implement NEP-17 standard.");
+
+                // Prevent NEO and GAS from being in extraTokens
+                if (contractHash == NEO.Hash || contractHash == GAS.Hash)
+                    throw new InvalidOperationException($"NEO and GAS should not be included in extraTokens. They are automatically processed.");
+
+                // Prevent duplicate tokens
+                if (!validatedTokens.Add(contractHash))
+                    throw new InvalidOperationException($"Duplicate token {contractHash} in extraTokens.");
+            }
+
+            // Transfer funds, NEO, GAS and extra NEP17 tokens
+
+            foreach (var contractHash in validatedTokens)
+            {
+                engine.CallContract(contractHash, "balanceOf", CallFlags.ReadOnly, new VM.Types.Array(engine.ReferenceCounter, [account.ToArray()]));
+
+                var balance = engine.Pop<Integer>().GetInteger();
+
+                if (balance > 0)
+                {
+                    // Mock account witness in CheckWitnessInternal
+
+                    var state = engine.CurrentContext!.GetState<ExecutionContextState>();
+                    var bak = state.NativeCallingScriptHash;
+                    state.NativeCallingScriptHash = account;
+
+                    try
+                    {
+                        engine.CallContract(contractHash, "transfer", CallFlags.All,
+                            new VM.Types.Array(engine.ReferenceCounter,
+                            [account.ToArray(), NativeContract.Treasury.Hash.ToArray(), balance, StackItem.Null]));
+                    }
+                    catch { throw; }
+                    finally
+                    {
+                        // Reset witnesses
+
+                        state.NativeCallingScriptHash = bak;
+                    }
+                }
+            }
+
+            // Remove and notify
+
+            engine.SnapshotCache.Delete(key);
```

Comment:
Consider removing key *before* `transfer` call? To avoid potential reentrance issues like it was with OnNEP17Payment.